performance-tools/performance-cluster/scripts/kyma-gke-cluster.sh

```diff
@@ -0,0 +1,116 @@
+#  - CLOUDSDK_COMPUTE_REGION - GCP compute region. Ex. "europe-west3"
+#  - CLOUDSDK_COMPUTE_ZONE Ex. "europe-west3-a"
+#  - CLOUDSDK_DNS_ZONE_NAME - GCP zone name (not its DNS name!). Ex. ""build-kyma""
+#  - GOOGLE_APPLICATION_CREDENTIALS - GCP Service Account key file path.
+#    Ex. "/etc/credentials/sa-gke-kyma-integration/service-account.json"
+#
+#  - KYMA_ARTIFACTS_BUCKET: GCP bucket. It defines the environment variable for the Kyma's artifact bucket. Ex. gs://kyma-prow-artifacts
+#  - BOT_GITHUB_TOKEN: Bot github token used for API queries. Ex. "/etc/credentials/kyma-bot-github-ssh/key.pem"
+#  - BOT_GITHUB_EMAIL It sets the environment variables for the email of the bot account.
+#  - BOT_GITHUB_USER It sets the environment variables for the name of the bot account.
+#  - DOCKER_IN_DOCKER_ENABLED true
+#  - MACHINE_TYPE (optional): GKE machine type
+#  - CLUSTER_VERSION (optional): GKE cluster version
+#
+# Permissions: In order to run this script you need to use a service account with permissions equivalent to the following GCP roles:
+#  - Compute Admin
+#  - Kubernetes Engine Admin
+#  - Kubernetes Engine Cluster Admin
+#  - DNS Administrator
+#  - Service Account User
+#  - Storage Admin
+#  - Compute Network Admin
+
+set -o errexit
+set -o pipefail  # Fail a pipe if any sub-command fails.
+discoverUnsetVar=false
+
+for var in INPUT_CLUSTER_NAME DOCKER_PUSH_REPOSITORY DOCKER_PUSH_DIRECTORY KYMA_PROJECT_DIR CLOUDSDK_CORE_PROJECT CLOUDSDK_COMPUTE_REGION CLOUDSDK_COMPUTE_ZONE CLOUDSDK_DNS_ZONE_NAME GOOGLE_APPLICATION_CREDENTIALS KYMA_ARTIFACTS_BUCKET BOT_GITHUB_TOKEN BOT_GITHUB_EMAIL BOT_GITHUB_USER DOCKER_IN_DOCKER_ENABLED; do
+    if [ -z "${!var}" ] ; then
+        echo "ERROR: $var is not set"
+        discoverUnsetVar=true
+    fi
+done
+if [ "${discoverUnsetVar}" = true ] ; then
+    exit 1
+fi
+
+
+export KYMA_SOURCES_DIR="${KYMA_PROJECT_DIR}/kyma"
+export KYMA_SCRIPTS_DIR="${KYMA_SOURCES_DIR}/installation/scripts"
+export TEST_INFRA_SOURCES_DIR="${KYMA_PROJECT_DIR}/test-infra"
+export TEST_INFRA_PERFORMANCE_TOOLS_CLUSTER_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/performance-tools/performance-cluster/scripts/helpers"
+
+export GCLOUD_PROJECT_NAME="${CLOUDSDK_CORE_PROJECT}"
+export GCLOUD_COMPUTE_ZONE="${CLOUDSDK_COMPUTE_ZONE}"
+export GCLOUD_SERVICE_KEY_PATH="${GOOGLE_APPLICATION_CREDENTIALS}"
+export BUILD_TYPE="master"
+
+readonly REPO_OWNER="kyma-project"
+readonly REPO_NAME="kyma"
+readonly CURRENT_TIMESTAMP=$(date +%Y%m%d)
+readonly STANDARIZED_NAME=$(echo "${INPUT_CLUSTER_NAME}" | tr "[:upper:]" "[:lower:]")
+readonly DNS_SUBDOMAIN="${STANDARIZED_NAME}"
+export GCLOUD_NETWORK_NAME="kyma-performance-cluster-net"
+export GCLOUD_SUBNET_NAME="kyma-performance-cluster-subnet"
+
+export STANDARIZED_NAME
+export REPO_OWNER
+export REPO_NAME
+export CURRENT_TIMESTAMP
+
+# shellcheck disable=SC1090
+source "${TEST_INFRA_PERFORMANCE_TOOLS_CLUSTER_SCRIPTS}/library.sh"
```

Comment:
First step should be:
- Clone Kyma repository (missing)
  - Use user defined Kyma repository and branch, if provided. Otherwise, default can be: kyma-project/kyma:master

Reply:
This has been updated. Now we clone the master. For development flow we assume local kyma installation and also the repo is on the right branch.